.github/workflows/engine.yml

```
@@ -0,0 +1,92 @@
+name: engine
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+permissions:
+  # Allow checks read
+  checks: write
+  # Allow repo checkout
+  contents: write
+  # Allow PRs write
+  pull-requests: write
+  # Allow issues write
+  issues: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Probably worth having this cancel old builds

```suggestion
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```